File: UWP/Shiba/CommonProperty/GridProperty.cs
using Windows.UI.Xaml.Controls;
using Shiba.Controls;
using NativeView = Windows.UI.Xaml.FrameworkElement;
using NativeBinding = Windows.UI.Xaml.Data.Binding;
using NativeProperty = Windows.UI.Xaml.DependencyProperty;
using NativeThickness = Windows.UI.Xaml.Thickness;
using NativeViewGroup = Windows.UI.Xaml.Controls.Panel;


namespace Shiba.CommonProperty
{
    public class GridProperty : AbsCommonProperty<ShibaObject>
    {
        public override string Name { get; } = "grid";

        public override void SetValue(ShibaObject shibaObject, View targetShibaView, NativeView element, NativeViewGroup parent)
        {
            var row = shibaObject.Get<int>("row");
            var column = shibaObject.Get<int>("column");
            var rowSpan = shibaObject.Get<int>("rowSpan");
            var columnSpan = shibaObject.Get<int>("columnSpan");
            if (row != default) Grid.SetRow(element, row);

            if (column != default) Grid.SetColumn(element, column);

            if (rowSpan != default) Grid.SetRowSpan(element, rowSpan);

            if (columnSpan != default) Grid.SetColumnSpan(element, columnSpan);
        }
    }
}

File: Android/shiba/src/main/java/org/shibajs/shiba/commonProperty/GridProperty.kt
package org.shibajs.shiba.commonProperty

import android.view.ViewGroup
import org.shibajs.shiba.NativeView
import org.shibajs.shiba.type.ShibaObject

class GridProperty(override val name: String = "grid") : AbsCommonProperty<ShibaObject>() {
    override fun setValue(value: ShibaObject, nativeView: NativeView, parent: ViewGroup) {
        val row = value["row"]?.toString()?.toDouble() ?: 0.0
        val column = value["column"]?.toString()?.toDouble() ?: 0.0
        val rowSpan = value["rowSpan"]?.toString()?.toDouble() ?: 1.0
        val columnSpan = value["columnSpan"]?.toString()?.toDouble() ?: 1.0
        val layoutParams = androidx.gridlayout.widget.GridLayout.LayoutParams(nativeView.layoutParams)
        layoutParams.rowSpec = androidx.gridlayout.widget.GridLayout.spec(row.toInt(), rowSpan.toInt())
        layoutParams.columnSpec = androidx.gridlayout.widget.GridLayout.spec(column.toInt(), columnSpan.toInt())
        nativeView.layoutParams = layoutParams
    }

}